Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksProducerFilter.cc

```diff
@@ -0,0 +1,211 @@
+    mutable std::atomic<unsigned int> nAll_, nGood_;
+  };
+}
+
+class AlCaIsoTracksProducerFilter : public edm::stream::EDFilter<edm::GlobalCache<AlCaIsoTracksProdFilter::Counters> > {
+public:
+  explicit AlCaIsoTracksProducerFilter(edm::ParameterSet const&,
+				       const AlCaIsoTracksProdFilter::Counters* count);
+  ~AlCaIsoTracksProducerFilter();
+    
+  static std::unique_ptr<AlCaIsoTracksProdFilter::Counters> initializeGlobalCache(edm::ParameterSet const& iConfig) {
+    return std::make_unique<AlCaIsoTracksProdFilter::Counters>();
+  }
+
+  virtual bool filter(edm::Event&, edm::EventSetup const&) override;
+  virtual void endStream() override;
+  static  void globalEndJob(const AlCaIsoTracksProdFilter::Counters* counters);
+  static  void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+private:
+  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  
+  // ----------member data ---------------------------
+  HLTConfigProvider             hltConfig_;
+  std::vector<std::string>      trigNames_, HLTNames_;
+  unsigned int                  nRun_, nAll_, nGood_;
+  edm::EDGetTokenT<trigger::TriggerEvent>  tok_trigEvt_;
+  edm::EDGetTokenT<edm::TriggerResults>    tok_trigRes_;
+  std::string                   processName_;
+  edm::InputTag                 triggerEvent_, theTriggerResultsLabel_;
+
+};
+
+AlCaIsoTracksProducerFilter::AlCaIsoTracksProducerFilter(const edm::ParameterSet& iConfig, const AlCaIsoTracksProdFilter::Counters* count) :
+  nRun_(0), nAll_(0), nGood_(0) {
+  //now do what ever initialization is needed
+  trigNames_                          = iConfig.getParameter<std::vector<std::string> >("Triggers");
+  processName_                        = iConfig.getParameter<std::string>("ProcessName");
+  triggerEvent_                       = iConfig.getParameter<edm::InputTag>("TriggerEventLabel");
+  theTriggerResultsLabel_             = iConfig.getParameter<edm::InputTag>("TriggerResultLabel");
+
+  // define tokens for access
+  tok_trigEvt_  = consumes<trigger::TriggerEvent>(triggerEvent_);
+  tok_trigRes_  = consumes<edm::TriggerResults>(theTriggerResultsLabel_);
+
+  edm::LogInfo("HcalIsoTrack") << "Use process name " << processName_
+			       << " Labels " << triggerEvent_ << " and "
+			       << theTriggerResultsLabel_ << " selecting "
+			       << trigNames_.size() << " triggers\n";
+#ifdef EDM_ML_DEBUG
+  std::cout << "HcalIsoTrack: Use process name " << processName_
+	    << " Labels " << triggerEvent_ << " and "
+	    << theTriggerResultsLabel_ << " selecting "
+	    << trigNames_.size() << " triggers" << std::endl;
+#endif
+  for (unsigned int k=0; k<trigNames_.size(); ++k) {
+    edm::LogInfo("HcalIsoTrack") << "Trigger[" << k << "] " << trigNames_[k]
+				 << std::endl;
+#ifdef EDM_ML_DEBUG
+    std::cout << "HcalIsoTrack: Trigger[" << k << "] " << trigNames_[k]
```

Comment:
@bsunanda Is there any use of these cout statements now? There are few of them in this code. Although I think this ifdef won't let it print since it is defined without any initialisation. Is it false by default? Can you please confirm?